### PR TITLE
Add GetTrajectoryNodePoses() to gRPC service

### PR DESCRIPTION
**Pull request by cschuet**

### CODE REVIEW

File: cartographer_grpc/handlers/get_trajectory_node_poses_handler.h [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+class GetTrajectoryNodePosesHandler
+    : public framework::RpcHandler<google::protobuf::Empty,
+                                   proto::GetTrajectoryNodePosesResponse> {
+ public:
+  void OnRequest(const google::protobuf::Empty& request) override {
+    auto node_poses = GetContext<MapBuilderServer::MapBuilderContext>()
+                          ->map_builder()
+                          .pose_graph()
+                          ->GetTrajectoryNodePoses();
+
```
Comment (gaschler) [inline cartographer_grpc/handlers/get_trajectory_node_poses_handler.h]: drop empty line

File: cartographer_grpc/handlers/get_trajectory_node_poses_handler.h [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    auto response = cartographer::common::make_unique<
+        proto::GetTrajectoryNodePosesResponse>();
+    for (const auto& node_id_pose : node_poses) {
```
Comment (gaschler) [inline cartographer_grpc/handlers/get_trajectory_node_poses_handler.h]: mutable_poses()->Reserve?

File: cartographer_grpc/handlers/get_trajectory_node_poses_handler.h (hunk 1 above)
Comment (cschuet) [reply cartographer_grpc/handlers/get_trajectory_node_poses_handler.h]: Done.

**Comment by cschuet** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user cschuet. Merge queue now has a length of 1.

**Pull request merged**